src/backend/distributed/utils/colocation_utils.c

```diff
@@ -133,14 +122,31 @@ MarkTablesColocated(Oid sourceRelationId, Oid targetRelationId)
 		uint32 shardCount = ShardIntervalCount(sourceRelationId);
 		uint32 shardReplicationFactor = TableShardReplicationFactor(sourceRelationId);
 
+		Var *sourceDistributionColumn = PartitionKey(sourceRelationId);
+		Oid sourceDistributionColumnType = sourceDistributionColumn->vartype;
+
 		sourceColocationId = CreateColocationGroup(shardCount, shardReplicationFactor,
 												   sourceDistributionColumnType);
 		UpdateRelationColocationGroup(sourceRelationId, sourceColocationId);
 	}
 
+	targetColocationId = TableColocationId(targetRelationId);
+
 	/* finally set colocation group for the target relation */
 	UpdateRelationColocationGroup(targetRelationId, sourceColocationId);
 
+	/* if there is not any remaining table in the colocation group, delete it */
+	if (targetColocationId != INVALID_COLOCATION_ID)
+	{
+		List *colocatedTableList = ColocationGroupTableList(targetColocationId);
+		int colocatedTableCount = list_length(colocatedTableList);
+
+		if (colocatedTableCount == 0)
+		{
+			DeleteColocationGroup(targetColocationId);
```

Comment:
We delete colocation groups if we drop the last table in the colocation group. The line is here: https://github.com/citusdata/citus/blob/master/src/backend/distributed/citus--6.0-10--6.0-11.sql#L88

I wanted to have a consistent behavior. We should either keep both or remove both. What do you think?